src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -399,9 +456,7 @@ async def create_pex(
         *request.additional_args,
     ]
 
-    if request.internal_only:
-        # This will result in a faster build, but worse compatibility at runtime.
-        argv.append("--use-first-matching-interpreter")
```

Comment:
N.B.: This flag is no longer used. I plan to deprecate it over in PEX and remove - it has never made much sense over there.